test/bin/scenario.sh

```diff
@@ -247,37 +247,77 @@ wait_for_greenboot() {
     local -r vmname="${1}"
     local -r ip="${2}"
 
+    start_junit
+    trap "close_junit" EXIT
+
     if "${SKIP_GREENBOOT}"; then
         echo "Skipping greenboot check"
+        record_junit "${vmname}" "greenboot-check" "SKIPPED"
         return 0
     fi
 
     echo "Waiting ${VM_BOOT_TIMEOUT} for greenboot on ${vmname} to complete"
 
     local -r start_time=$(date +%s)
+    local -r ssh_cmd="ssh -oConnectTimeout=10 -oBatchMode=yes -oStrictHostKeyChecking=accept-new redhat@${ip}"
+    local -r kube_opt="--kubeconfig /var/lib/microshift/resources/kubeadmin/kubeconfig"
+    local retry_count=2
     while [ $(( $(date +%s) - start_time )) -lt "${VM_BOOT_TIMEOUT}" ] ; do
-        if ssh -oConnectTimeout=10 -oBatchMode=yes -oStrictHostKeyChecking=accept-new "redhat@${ip}" \
-                "sudo journalctl -n 5 -u greenboot-healthcheck; \
-                 systemctl show --property=SubState --value greenboot-healthcheck | grep -w exited" ; then
+        local svc_state
+
+        svc_state="$(${ssh_cmd} systemctl show --property=SubState --value greenboot-healthcheck || true)"
+        if [ "${svc_state}" = "exited" ] ; then
+            record_junit "${vmname}" "greenboot-check" "OK"
             return 0
         fi
+
+        # Print the last log and check for terminal failure
+        ${ssh_cmd} "sudo journalctl -n 10 -u greenboot-healthcheck" || true
```

Comment:
This looks like will print the logs on each loop-body execution when svc isn't neither `exited` nor `failed`. Should this be inside `if` below?

Reply:
In the original code, it was printed unconditionally on every iteration to show greenboot progress. I guess, I can change this, but it's not how it was

Reply:
Ah, right, ack. I though this is something extra